src/openms/source/ANALYSIS/OPENSWATH/MRMScoring.cpp

```diff
@@ -348,37 +308,39 @@ namespace OpenSwath
 
     std::vector<double> MRMScoring::calcSeparateXcorrContrastCoelutionScore()
     {
-      OPENSWATH_PRECONDITION(xcorr_contrast_matrix_max_peak_.rows() > 0 && xcorr_contrast_matrix_max_peak_.cols() > 1, "Expect cross-correlation matrix of at least 1x2");
+      OPENSWATH_PRECONDITION(xcorr_contrast_matrix_.rows() > 0 && xcorr_contrast_matrix_.cols() > 1, "Expect cross-correlation matrix of at least 1x2");
 
       std::vector<double > deltas;
-      for (std::size_t i = 0; i < xcorr_contrast_matrix_max_peak_.rows(); i++)
+      for (std::size_t i = 0; i < xcorr_contrast_matrix_.rows(); i++)
       {
         double deltas_id = 0;
-        for (std::size_t  j = 0; j < xcorr_contrast_matrix_max_peak_.cols(); j++)
+        for (std::size_t  j = 0; j < xcorr_contrast_matrix_.cols(); j++)
         {
           // first is the X value (RT), should be an int
-          deltas_id += xcorr_contrast_matrix_max_peak_.getValue(i, j);
+          auto x = Scoring::xcorrArrayGetMaxPeak(xcorr_contrast_matrix_.getValue(i, j));
+          deltas_id += std::abs(x->first);
 #ifdef MRMSCORING_TESTING
           std::cout << "&&_xcoel append " << xcorr_contrast_matrix_max_peak_getValue(i, j) << std::endl;
 #endif
         }
-        deltas.push_back(deltas_id / xcorr_contrast_matrix_max_peak_.cols());
+        deltas.push_back(deltas_id / xcorr_contrast_matrix_.cols());
       }
 
       return deltas;
     }
 
     double MRMScoring::calcXcorrPrecursorCoelutionScore()
     {
-      OPENSWATH_PRECONDITION(xcorr_precursor_matrix_max_peak_.rows() > 1, "Expect cross-correlation matrix of at least 2x2");
+      OPENSWATH_PRECONDITION(xcorr_precursor_matrix_.rows() > 1, "Expect cross-correlation matrix of at least 2x2");
 
       std::vector<int> deltas;
-      for (std::size_t i = 0; i < xcorr_precursor_matrix_max_peak_.rows(); i++)
+      for (std::size_t i = 0; i < xcorr_precursor_matrix_.rows(); i++)
       {
-        for (std::size_t  j = i; j < xcorr_precursor_matrix_max_peak_.rows(); j++)
+        for (std::size_t  j = i; j < xcorr_precursor_matrix_.rows(); j++)
         {
           // first is the X value (RT), should be an int
-          deltas.push_back(xcorr_precursor_matrix_max_peak_.getValue(i, j));
+          auto x = Scoring::xcorrArrayGetMaxPeak(xcorr_precursor_matrix_.getValue(i, j));
+          deltas.push_back(std::abs(x->first));
```

Comment:
instead of first creating a `deltas` vector, one could simply call
`msc(std::abs(x->first))` within the loop?

Reply:
Based on the implementation of `msc` it's not doable to call it on every single element rather than it needs to keep track of the whole vector to compute the mean and deviation.

Reply:
why?
```
  std::vector<double> deltas(1000);
  std::iota(deltas.begin(), deltas.end(), 0);
  
  // before
  OpenSwath::mean_and_stddev msc;
  msc = std::for_each(deltas.begin(), deltas.end(), msc);
  std::cout << msc.mean() << " " << msc.sample_stddev() << "\n\n\n";

  // after
  OpenSwath::mean_and_stddev msc2;
  for (auto val : deltas) {
    msc2(val);
  }
  std::cout << msc2.mean() << " " << msc2.sample_stddev() << "\n\n\n";
```

Reply:
Ahhh 😅😅 Thanks for your comments I changed all occurrences of it and will push again soon!